src/Offcanvas.php

```diff
@@ -4,235 +4,554 @@
+    public function placement(OffcanvasPlacement|string $placement): self
     {
         $new = clone $this;
-        $new->bodyOptions = $options;
+        $new->placement = $placement;
 
         return $new;
     }
 
-    public function begin(): string
+    /**
+     * Sets the responsive size.
+     *
+     * @param Responsive|string $size The responsive size.
+     *
+     * @return self A new instance with the specified responsive size setting.
+     */
+    public function responsive(Responsive|string $size): self
     {
-        parent::begin();
+        $size = $size instanceof Responsive ? $size->value : $size;
 
-        $options = $this->options;
-        $bodyOptions = $this->bodyOptions;
-        $tag = ArrayHelper::remove($options, 'tag', 'div');
-        $bodyTag = ArrayHelper::remove($bodyOptions, 'tag', 'div');
+        $new = clone $this;
+        $new->responsive = match ($size) {
+            'sm', 'md', 'lg', 'xl', 'xxl' => $size,
+            default => '',
+        };
 
-        Html::addCssClass($options, ['widget' => 'offcanvas', 'placement' => $this->placement]);
-        Html::addCssClass($bodyOptions, ['widget' => 'offcanvas-body']);
+        return $new;
+    }
 
-        $options['id'] = $this->getId();
-        $options['tabindex'] = -1;
+    /**
+     * Sets whether is scrollable.
+     *
+     * @return self A new instance with the specified scrollable setting.
+     *
+     * @link https://getbootstrap.com/docs/5.3/components/offcanvas/#body-scrolling
+     */
+    public function scrollable(): self
+    {
+        $new = clone $this;
+        $new->scrollable = true;
 
-        if ($this->title !== null && $this->title !== '') {
-            if (isset($this->titleOptions['id'])) {
-                $options['aria-labelledby'] = $this->titleOptions['id'];
-            } elseif ($options['id']) {
-                $options['aria-labelledby'] = $options['id'] . '-title';
-            }
-        }
+        return $new;
+    }
 
-        if ($this->scroll) {
-            $options['data-bs-scroll'] = 'true';
-        }
+    /**
+     * Sets whether is visible.
+     *
+     * @param bool $show Whether is visible.
+     *
+     * @return self A new instance with the specified visibility setting.
+     */
+    public function show(): self
+    {
+        $new = clone $this;
+        $new->show = true;
 
-        if ($this->withoutBackdrop) {
-            $options['data-bs-backdrop'] = 'false';
-        }
+        return $new;
+    }
 
-        if ($this->theme) {
-            $options['data-bs-theme'] = $this->theme;
+    /**
+     * Sets the theme.
+     *
+     * @param string $theme The theme. If an empty string, the theme will be removed.
+     * Valid values are `dark` and `light`.
+     *
+     * @return self A new instance with the specified theme.
+     *
+     * @link https://getbootstrap.com/docs/5.3/components/offcanvas/#dark-offcanvas
+     */
+    public function theme(string $theme): self
```

Comment:
The name is a bit confusing since there's widget's theme.

Reply:
In the end i will change it in all widgets for another name, maybe i will add light and dark, or change it by mode, since you can literally add any color to the mode.